driver/src/main/java/org/neo4j/driver/Session.java

```diff
@@ -47,6 +47,13 @@
      */
     Transaction newTransaction();
 
+    /**
+     * Detect whether the session is currently open or not.
+     *
+     * @return true if the session is open, false if it has been {@link #close() closed}.
+     */
+    boolean isOpen();
```

Comment:
I think all this should move up to StatementRunner, valid for transactions, too, right?